ServerCore/Pages/Events/Mailer.cshtml.cs

```diff
@@ -42,11 +48,16 @@ public MailerModel(PuzzleServerContext serverContext, UserManager<IdentityUser>
         {
         }
 
-        public async Task<IActionResult> OnGetAsync(MailGroup group, int? teamId)
+        public async Task<IActionResult> OnGetAsync(MailGroup group, int? puzzleId, int? teamId)
```

Comment:
Could this be a different method? As far as I can tell, providing both puzzleId and teamId or combining puzzleId with any MailGroup other than NonSolvers doesn't work.

Reply:
I don't know how

Reply:
Added validation